src/Illuminate/Database/Connection.php

```diff
@@ -343,6 +345,24 @@ public function select($query, $bindings = [], $useReadPdo = true)
     }
 
     /**
+     * Bind values to their parameters in the given statement.
+     *
+     * @param  array  $bindings
+     * @param  \PDOStatement $statement
+     * @return void
+     */
+    public function bindValues($bindings, $statement)
+    {
+        foreach ($bindings as $key => $value) {
+            $statement->bindValue(
+                is_string($key) ? $key : $key + 1,
+                $value,
+                is_numeric($value) ? PDO::PARAM_INT : PDO::PARAM_STR
```

Comment:
`filter_var($value, FILTER_VALIDATE_INT) !== false` will validate for a real integer type. is_numeric could be a string / float.